src/transport/smtp/client/async_connection.rs

```diff
@@ -62,7 +62,27 @@ impl AsyncSmtpConnection {
 
     /// Connects to the configured server
     ///
+    /// If `tls_parameters` is not `None`, then the connection will use Implicit TLS (sometimes
+    /// referred to as `SMTPS`). See also [`AsyncSmtpConnection::starttls`].
+    ///
+    /// If `local_addres` is not `None`, then the address provided shall be used to bind the
+    /// connection to a specific local address using [`tokio1_crate::net::TcpSocket::bind`].
```

Comment:
The double negation mixed me up when I first read it. This should be a bit clearer

```suggestion
    /// If `tls_parameters` is `Some`, then the connection will use Implicit TLS (sometimes
    /// referred to as `SMTPS`). See also [`AsyncSmtpConnection::starttls`].
    ///
    /// If `local_addres` is `Some`, then the address provided shall be used to bind the
    /// connection to a specific local address using [`tokio1_crate::net::TcpSocket::bind`].
```